src/extension.ts

```diff
@@ -159,22 +167,32 @@ export function activate(context: ExtensionContext) {
         revealOutputChannelOn: CONFIGURATION.revealOutputChannelOn,
         initializationOptions: { omitInitBuild: true },
     };
-
+    
     // Create the language client and start the client.
-    const lc = new LanguageClient('Rust Language Server', serverOptions, clientOptions);
+    lc = new LanguageClient('Rust Language Server', serverOptions, clientOptions);
     lcOutputChannel = lc.outputChannel;
-
-    diagnosticCounter(lc);
-    registerCommands(lc, context);
-    activateTaskProvider();
-
+    
+    let runningDiagnostics = 0;
```

Comment:
Why do you want to inline this code?

Reply:
It didn't seem to be obviously a reusable function --- calling it repeatedly doesn't make sense?

Reply:
It's more for self-documentation - I found it made the `activate` function easy to read by factoring out self-contained bits of functionality. Inlining them made for a long, hard to grok function.